### PR TITLE
Add new.target transform

### LINKED ISSUE #1088: Class support should include `new.target`
Expected:

``` js
class Foo { constructor() { console.log(new.target.name); } }
var x = new Foo(); // "Foo"
```

and...

``` js
class Bar extends Foo { constructor() { super() } }
var y = new Bar(); // "Bar"
```

### PR DESCRIPTION
**Pull request by jridgewell**
| Q                        | A 
| ------------------------ | ---
| Patch: Bug Fix?          | No
| Major: Breaking Change?  | No
| Minor: New Feature?      | Yes
| Deprecations?            | No
| Spec Compliancy?         | Yes?
| Tests Added/Pass?        | Yes
| Fixed Tickets            | Fixes #1088 
| License                  | MIT
| Doc PR                   |
| Dependency Changes       | 

Transforms `new.target` into a (correct in most cases) `this.constructor` expression.

### CODE REVIEW

**Comment by jridgewell** [comment]
I realized we could transform all Class uses of `new.target`, because the constructors have to be called as constructors. Now the only real edge cases are using `new.target` before `super()`, and ES5-style function class inheritance.

File: packages/babel-plugin-transform-new-target/test/fixtures/general/arrow/expected.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  var _newtarget = this instanceof Foo ? this.constructor : void 0;
+
+  const a = function () {
+    _newtarget;
+  };
+}
+
+class Bar {
+  constructor() {
+    var _newtarget2 = this.constructor;
```
Comment (bergus) [inline packages/babel-plugin-transform-new-target/test/fixtures/general/arrow/expected.js]: Not sure why this variable should have a `2` suffix?

Comment (jridgewell) [inline packages/babel-plugin-transform-new-target/test/fixtures/general/arrow/expected.js]: That's how babel `generateUidIdentifier` works, it's unique globally.
